### PR TITLE
Correlate cli to service

### PR DESCRIPTION
**Pull request by tevoinea**
## Summary of the Pull Request

_What is this about?_

TLDR DEMO:
```shell
$ onefuzz jobs list -vv
DEBUG:onefuzz:list jobs
DEBUG:backend:request GET https://example.azurewebsites.net/api/jobs {'job_id': None, 'state': ['init', 'enabled'], 'task_state': None, 'with_tasks': None}
DEBUG:backend:OneFuzz CorrelationId: 5ad369b45a46ad29f4ec841149eda287
[]
```

In AppInsights TransactionSearch, paste the CorrelationId from the CLI:
image
Results screenshot was trimmed for brevity.

## PR Checklist
* [ ] Applies to work item: #xxx
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

_What does this include?_

1. The OneFuzz service accepts the OpenTelemetry standard `traceparent` header
2. The OneFuzz service includes the `traceparent` header in all responses
3. The OneFuzz CLI creates a new trace context and includes the `traceparent` header in requests to the service

Our logging/tracing/telemetry still happens through ILogger just as before. We're just connecting traces together using `traceparent`.

The CLI isn't _capturing_ or submitting any new telemetry, we're just making CLI invocations traceable in our service.

## Validation Steps Performed

_How does someone test & validate?_

check_pr running.

### CODE REVIEW

File: src/ApiService/ApiService/Functions/Containers.cs [hunk 1]
```diff
@@ -107,7 +107,7 @@ public Async.Task<HttpResponseData> Run([HttpTrigger(AuthorizationLevel.Anonymou
                 context: post.Name.String);
         }
 
-        return await RequestHandling.Ok(
+        return await new RequestHandling(_log).Ok(
```
Comment (Porges) [inline src/ApiService/ApiService/Functions/Containers.cs:110]: we should probably make this dependency injected

File: src/ApiService/ApiService/Program.cs [hunk 2]
```diff
@@ -32,7 +32,24 @@ public class LoggingMiddleware : IFunctionsWorkerMiddleware {
                 //if correlation ID is available in Queue message
                 //log.ReplaceCorrelationId(Guid from request)
 
-                log.ReplaceCorrelationId(Guid.NewGuid());
+                var externalRequestWithOtel = false;
+                var requestData = await context.GetHttpRequestDataAsync();
+                if (requestData != null && requestData.Headers.Contains("traceparent")) {
```
Comment (tevoinea) [inline src/ApiService/ApiService/Program.cs:37]: Look at context.TraceContext.TraceParent

File: src/ApiService/ApiService/Program.cs [hunk 3]
```diff
@@ -36,27 +35,26 @@ public class LoggingMiddleware : IFunctionsWorkerMiddleware {
         public async Async.Task Invoke(FunctionContext context, FunctionExecutionDelegate next) {
             //https://learn.microsoft.com/en-us/azure/azure-monitor/app/custom-operations-tracking#applicationinsights-operations-vs-systemdiagnosticsactivity
             using var activity = OneFuzzLogger.Activity;
-            _ = activity.Start();
-            string correlationId = Guid.NewGuid().ToString();
-
-            if (await context.GetHttpRequestDataAsync() is HttpRequestData requestData) {
```
Comment (stishkin) [inline src/ApiService/ApiService/Program.cs:42]: why did you remove getting Correlation ID from the http header ? 

request comes it -> we'd get Correlation ID from the request, and we can use that in the rest of the execution flow.

File: src/ApiService/ApiService/Program.cs [hunk 4]
```diff
@@ -36,27 +35,26 @@ public class LoggingMiddleware : IFunctionsWorkerMiddleware {
...
+            _ = activity.AddTag(OneFuzzLogger.SpanId, activity.SpanId);
             _ = activity.AddTag("FunctionId", context.FunctionId);
             _ = activity.AddTag("InvocationId", context.InvocationId);
 
             await next(context);
 
-            context.GetHttpResponseData()?.Headers.Add("X-Correlation-ID", correlationId);
+            var response = context.GetHttpResponseData();
+
+            response?.Headers.Add("traceparent", $"00-{activity.TraceId}-{activity.SpanId}-{(int)activity.ActivityTraceFlags:D2}");
```
Comment (stishkin) [inline src/ApiService/ApiService/Program.cs:57]: I think Activity.ID already has the format that you are building by hand there

File: src/ApiService/ApiService/Program.cs (hunk 3 above)
Comment (stishkin) [reply src/ApiService/ApiService/Program.cs:42]: If execution flow starts with CLI, how does CLI set the initial correlation ID ?

**Review by stishkin — APPROVED** [review]
